lib.php

```diff
@@ -81,52 +84,12 @@ public static function has_config() {
 
     public static function get_config_options() {
 
-        // Get default config;
-        $defaultconfig = get_objectfs_config();
+      $defaultconfig = self::get_default_config();
 
-        if (isset($_POST)) {
-            foreach ($_POST as $key => $value) {
-                foreach ($defaultconfig as $key1 => $value1) {
-                    if ($key == $key1) {
-                        $defaultconfig->$key1 = $value;
-                    }
-                }
-            }
-        }
+      $connection = self::check_s3_connection($defaultconfig);
```

Comment:
These checks should be updated so they're dynamic - I.E. only run on defined clients. The goal here is such that when we add another object client we just drag and drop it in - we shouldn't have to update this file at all.